networking_ccloud/tools/netbox_config_gen.py

```diff
@@ -80,21 +83,45 @@ def __init__(self, region, args, verbose=False, verify_ssl=False):
             self.netbox.http_session = requests.Session()
             self.netbox.http_session.verify = False
 
-    def _ensure_single_item(self, item_set: Iterable[Any], entity: str, identifier: str, attribute: str) -> Any:
+    @classmethod
+    def _ignore_filter(cls, items: Iterable[NbRecord]) -> Generator[NbRecord, None, None]:
```

Comment:
Why does this need to be a generator? Is that only so you don't need to fetch certain attributes twice and you can't make a list out of them beforehand? It makes the code flow a bit unintuitive to me in some places.

Reply:
This is likely going to take a `RecordSet` from `pynetbox` which is a generator already. I did not want to break with that concept as in some places, the netbox responses generator is taken and iterated over straight away. If I make this return a list, I break that generator concept. I know it's just minor memory allocation here and there, but still I would like to keep it.

Reply:
Okay, then let's keep it for now. We can still revisit this if we see the need for it.